goodtablesio/helpers/prepare.py

```diff
@@ -22,28 +22,29 @@ def prepare_job(job_conf_url, job_files):
         validation_conf (dict): Configuration object to be used by the
             validation task
 
-    This function is separate to make testing easier.
+    This function is pure to make testing easier.
+
     """
+    job_conf = _load_job_conf(job_base)
+    validate_job_conf(job_conf)
+    validation_conf = _make_validation_conf(job_base, job_files, job_conf)
+    return validation_conf
 
-    # Get base url and load job configuration
-    base_url = job_conf_url.rsplit('/', 1)[0]
-    yml_file = _load_file(job_conf_url)
-    if yml_file:
-        job_conf = yaml.load(yml_file)
 
-        # Validate job configuration
-        validate_job_conf(job_conf)
+# Internal
 
+
+def _load_job_conf(job_base):
+    url = '/'.join([job_base, 'goodtables.yml'])
+    text = _load_file(url)
+    if text is not None:
+        job_conf = yaml.load(text)
```

Comment:
What happens if it's not valid YML? Also we should [secure](http://pyyaml.org/wiki/PyYAMLDocumentation#LoadingYAML) this as we can be essentially loading anything from the internet. This can be a separate issue though